File: mario/mario.c
#include <stdio.h>
#include <cs50.h>

int get_height(void);
int calculate_spaces(int height);
int calculate_hashes(int n);
void print_pyramid(int h);

int main(void)
{
    int h = get_height();
    print_pyramid(h);
    
}

int get_height(void)
{
    int h;
    // Prompt user for height
    do
    {
        h = get_int("Height: ");
    } while (h < 1 || h > 8);
    return h;
}

int calculate_spaces(int height)
{
    // Calculate spaces needed
    int space = height - 1;
    return space;
}

int calculate_hashes(int n)
{
    // Calculate hashes needed
    int hash = n;
    return hash;
}

void print_pyramid(int height)
{
    int min = 1;
    int max = height;

    for (int i = 0; i < height; i++)
    {
        // Calculate Spaces and Hashes
        int s = calculate_spaces(max);
        int h = calculate_hashes(min);

        // Print Tower

        // Print Spaces
        for (int j = 0; j < s; j++)
        {
            printf(" ");
        }
        // Print left hashes
        for (int k = 0; k < h; k++)
        {
            printf("#");
        }
        // Print Gap
        printf("  ");
        // Print right hashes
        for (int k = 0; k < h; k++)
        {
            printf("#");
        }

        printf("\n");
        min++;
        max--;
    }
}

File: mario/mario.py
from cs50 import get_int


def main():
    # Validate height choice
    h = valid_height()

    # Print Pyramid
    print_pyramid(h)


def valid_height():
    while True:
        # Ask for height
        height = get_int("Height: ")
        # Make height between 1 and 8 inclusive
        if height >= 1 and height <= 8:
            break
    return height


def calculate_spaces(height):
    # Calculate spaces needed
    space = height - 1
    return space


def calculate_hashes(n):
    # Calculate hash
    hashes = n
    return hashes


def print_pyramid(height):
    min_value = 1
    max_value = height

    # Calculate Spaces and Hashes
    for i in range(height):
        # Calculate spaces and hashes
        s = calculate_spaces(max_value)
        h = calculate_hashes(min_value)

        # Print Pyramid
        # Print Spaces
        for j in range(s):
            print(" ", end="")
        # Print left hashes
        for k in range(h):
            print("#", end="")
        # Print Gap
        print("  ", end="")
        # Print Right hashes
        for k in range(h):
            print("#", end="")
        print()

        # Change values for each row
        min_value += 1
        max_value -= 1


main()
